### PR TITLE
feat: add nullif function

### PR DESCRIPTION
**Pull request by thisisnic**
This PR adds 2 conditional functions:  `if_else` and `nullif`.

### CODE REVIEW

File: extensions/functions_comparison.yaml [hunk 1]
```diff
@@ -107,4 +107,23 @@ scalar_functions:
           - value: fp64
         nullability: DECLARED_OUTPUT
         return: BOOLEAN
+  -
+    name: "if_else"
+    description: If true, return the first value, otherwise return the second.
```
Comment (thisisnic) [inline extensions/functions_comparison.yaml:112]: I'm a bit unsure about phrasing here; does it make sense to refer to the second and third parameters as "first" and "second" values? Perhaps not. Not sure how best to go about this.

**Comment by jvanstraten** [comment]
If-else is actually special-cased as an expression type, so there's no need for a function for it. See https://substrait.io/expressions/specialized_record_expressions/#if-expression

**Comment by thisisnic** [comment]
> If-else is actually special-cased as an expression type, so there's no need for a function for it. See https://substrait.io/expressions/specialized_record_expressions/#if-expression

Thanks, removed now.

**Review by jvanstraten — APPROVED** [review]
LGTM

**Pull request merged**